v1.7.0/longhorn-engine/rockcraft.yaml

```diff
@@ -0,0 +1,267 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Rockcraft definition for Longhorn engine image:
+# longhornio/longhorn-engine:v1.7.0
+
+name: longhorn-engine
+summary: Rock containing Longhorn engine component.
+description: |
+  Rock containing Longhorn engine component: https://github.com/longhorn/longhorn-engine
+  Aims to replicate the upstream official image: longhornio/longhorn-engine:v1.7.0
+license: Apache-2.0
+
+version: "v1.7.0"
+
+# NOTE(aznashwan): the base for the engine image is the Suse Linux Enterprise
+# Base Container Image (SLE BCE) Service Pack 6 which ships with Linux 6.4,
+# and is thus most comparable to 24.04:
+# https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L1
+base: ubuntu@24.04
+
+build-base: ubuntu@24.04
+platforms:
+  amd64:
+  arm64:
+
+
+services:
+  longhorn-engine:
+    startup: enabled
+    override: replace
+
+    # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L68-L70
+    command: /tini -- [ longhorn ]
+
+parts:
+
+  # NOTE(aznashwan): the longhorn binary is built within a Docker container
+  # which is set up by Rancher's Dapper tool: https://github.com/rancher/dapper
+  # The setup steps for the build container are contained within this Dockerfile:
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper
+  setup-build-env:
+    plugin: nil
+
+    build-packages:
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper#L30-L36
+      - cmake
+      - curl
+      - git
+      - less
+      - file
+      - gcc
+      - python3
+      - python3-pip
+      - python3-dev
+      - libkmod-dev
+      - libnl-3-dev
+      - linux-headers-6.8.0-41
+      - pkg-config
+      - psmisc
+      # - qemu-tools
+      - fuse
+      - librdmacm-dev
+      - rdmacm-utils
+      - libibverbs-dev
+      - xsltproc
+      - docbook-xsl
+      - ldp-docbook-xsl
+      - perl
+      - libconfig-general-perl
+      - libaio-dev
+      - libc6-dev
+      - sg3-utils
+      - iptables
+      - libltdl7
+      - libdevmapper-dev
+      - iproute2
+      - jq
+      - unzip
+      - zlib1g-dev
+      - rdma-core
+      - g++
+      - libopeniscsiusr-dev
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper#L25-L27
+      - autoconf
+      - libtool
+      - libunwind-dev
+
+    override-build: |
+      set -eux -o pipefail
+
+      # NOTE(aznashwan): pull and install libqcow2 for building the engine:
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper#L48-L53
+      curl -sSfL https://s3-us-west-1.amazonaws.com/rancher-longhorn/libqcow-alpha-20181117.tar.gz | tar xvzf - -C /usr/src
```

Comment:
I wonder if we could use libqcow from the ubuntu archive or build from source, I'm fine with having this as a start but we should look into updating this later on.

Reply:
We absolutely could, though I intentionally wanted to download and use the Rancher-vendored one for compatibility reasons in case they have added/changed any functionality on their fork.

The only thing that could prevent us using the Ubuntu-packaged lib is compatibility issues (the version in Noble is `20201213` vs Rancher's `20181117`), though I did NOT try using the .deb to confirm one way or the other...

Reply:
We can keep this as is for the start and keep it in mind, maybe adding a TODO is helpful.